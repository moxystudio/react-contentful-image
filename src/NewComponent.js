import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const NewComponent = ({ children, className }) => (
    <div className={ classNames('next-lib-template_container', className) }>
        { children }
    </div>
);

NewComponent.propTypes = {
    children: PropTypes.any.isRequired,
    className: PropTypes.string,
};

export default NewComponent;
