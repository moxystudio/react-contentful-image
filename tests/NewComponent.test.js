import React from 'react';
import { render } from '@testing-library/react';
import NewComponent from '../src/NewComponent';

const defaultProps = {
    children: <span>Hello</span>,
};

const renderWithProps = (props = {}) => render(<NewComponent { ...defaultProps } { ...props } />);

describe('NewComponent Component', () => {
    it('should render correctly', () => {
        const { getByText } = renderWithProps();

        expect(getByText('Hello')).toBeInTheDocument();
    });

    it('should render correctly with different children', () => {
        const { asFragment } = renderWithProps({
            children: <span>Foo Bar</span>,
        });

        expect(asFragment()).toMatchSnapshot();
    });
});
