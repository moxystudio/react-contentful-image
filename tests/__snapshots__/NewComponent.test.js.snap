// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComponent Component should render correctly with different children 1`] = `
<DocumentFragment>
  <div
    class="next-lib-template_container"
  >
    <span>
      Foo Bar
    </span>
  </div>
</DocumentFragment>
`;
